components/Hooks/useLocalStorageState.js

```diff
@@ -0,0 +1,18 @@
+import { useState, useEffect } from "react";
+
+export default function useLocalStorageState(key, initial) {
+  if (typeof window !== "undefined") {
+    // Try and get some values from localStorage:
+    // Set up defaults if there's nothing in localStorage (i.e. the values are null):
+    const [value, setValue] = useState(() => {
+      return JSON.parse(window.localStorage.getItem(key)) || initial;

```

Comment:
Interesting approach on updating and recognising the values from the initial local storage ✨